src/test/java/org/mongojack/internal/util/CalendarDeserializerTest.java

```diff
@@ -105,7 +105,7 @@ public void testNoEmbeddedObject() throws IOException {
     @Test
     public void testDateIsNull() throws IOException {
         when(jsonParser.getCurrentToken()).thenReturn(JsonToken.VALUE_NULL);
-        when(jsonParser.getCurrentTokenId()).thenReturn(JsonTokenId.ID_NULL);
+        when(jsonParser.currentTokenId()).thenReturn(JsonTokenId.ID_NULL);
```

Comment:
Can you explain why this was necessary?  Is it just required by Jackson 2.12?

Reply:
Because `StdDeserializer#_parseDate` started using `#currentTokenId` instead. https://github.com/FasterXML/jackson-databind/commit/0afc8223c344fcc0a634b7bd83b1861d668fc223#diff-416896884d1e0706562f38c160757ff53196c441ce86a4ca49923f7aebad6a36L496-R496

Without this change, we fall back to the default case and get `MismatchedInputException`.